### PR TITLE
Initial Elasticsearch store commit

### PR DESCRIPTION
**Pull request by ksdperera**
Elasticsearch store implementation uses Elasticsearch indexing document for underlying data storage. The events are converted to Elasticsearch index documents when the events are inserted into Elasticsearch store. Elasticsearch indexing documents are converted to Events when the documents are read from Elasticsearch indexes. Internally store connected with Elasticsearch server with The Elasticsearch Java High-Level REST Client library.

## Automation tests
 - Integration tests
   > Docker Test for DAL CRUD operations + Siddhi Contain Check:
```mvn verify -P local-elasticsearch -f component/pom.xml -Dskip.surefire.test=true'```

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,44 @@
-# siddhi-store-elasticsearch
\ No newline at end of file
+﻿siddhi-store-elasticsearch
+======================================
+
```
Comment (Niveathika92) [inline README.md]: README is not complete

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+site_name: Siddhi-Store-Elasticsearch
+extra_css:
```
Comment (Niveathika92) [inline mkdocs.yml]: Needs to be changed to support mkdocs theme update

Comment (ksdperera) [reply mkdocs.yml]: Added the fixes: a3ce67ba98728e883dcd53baf7f40b92ee61e788

File: README.md (hunk 1 above)
Comment (ksdperera) [reply README.md]: Added the fixes: a3ce67b

File: mkdocs.yml (hunk 2 above)
Comment (ksdperera) [reply mkdocs.yml]: adec20ea2c6499ccab87d812cb6a1dd0bc18c438

**Pull request merged**